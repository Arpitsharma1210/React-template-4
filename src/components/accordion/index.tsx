import React, { useState } from "react";
import {
  StyledTypography,
  AddIcon,
  CloseIcon,
} from "./styles";

import { Accordion as MuiAccordion, AccordionSummary, AccordionDetails } from "@mui/material";

interface CustomAccordionProps {
  title: string;
  contentComponent: React.ReactNode;
}

const Accordion: React.FC<CustomAccordionProps> = ({
  title,
  contentComponent,
}) => {
  const [expanded, setExpanded] = useState<boolean>(false);
  return (
    <MuiAccordion>
      <AccordionSummary onClick={() => setExpanded(!expanded)}>
        <StyledTypography expanded={expanded}>{title}</StyledTypography>
        {expanded ? <CloseIcon /> : <AddIcon />}
      </AccordionSummary>
      <AccordionDetails>{contentComponent}</AccordionDetails>
      </MuiAccordion>
  );
};

export default Accordion;
