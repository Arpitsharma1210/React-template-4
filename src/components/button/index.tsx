import React from "react";
import { StyledButtonTextWrapper } from "./styles";
import { Button } from "@mui/material";
interface Props {
  children?: any;
  variant?: "text" | "outlined" | "contained";
  bold?: boolean;
  onClick?: any;
  type?: "button" | "reset" | "submit";
  disabled?: boolean;
  label?: string;
  style?: React.CSSProperties;
}

const MuiButton: React.FC<Props> = ({
  children,
  variant,
  bold = false,
  onClick,
  type,
  disabled,
  label,
  style,
  ...props
}) => {
  const clickHandler = (e: any) => {
    if (onClick) {
      onClick(e);
    }
  };

  return (
    <Button
      onClick={clickHandler}
      variant={variant || "contained"}
      type={type || "button"}
      disabled={disabled}
      {...props}
    >
      <StyledButtonTextWrapper
        className={`bold_btnText_${bold}`}
        $bold={bold.toString()}
      >
        {children}
      </StyledButtonTextWrapper>
    </Button>
  );
};

export default MuiButton;
