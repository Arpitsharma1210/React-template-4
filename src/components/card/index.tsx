import React from "react";
import { Card as MaterialCard } from "@mui/material";

interface Props {
  children: JSX.Element | JSX.Element[];
}

const Card: React.FC<Props> = ({ children }) => {
  return (
    <MaterialCard >{children}</MaterialCard>
  );
};

export default Card;
