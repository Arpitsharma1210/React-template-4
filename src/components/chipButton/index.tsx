import React from "react";
import { StyledButton } from "./styles";

type ButtonColors =
  | "yellow"
  | "yellow2"
  | "green"
  | "blue"
  | "orange"
  | "red"
  | "purple"
  | "white"
  | "grey";

interface Props {
  color: ButtonColors;
  size?: "md" | "sm";
  children?: any;
  onClick?: any;
}

const ChipButton: React.FC<Props> = ({
  color,
  size = "md",
  children,
  onClick,
}) => {
  const clickHandler = (e: any) => {
    if (onClick) {
      onClick(e);
    }
  };

  return (
    <StyledButton onClick={clickHandler} size={size} color={color}>
      {children}
    </StyledButton>
  );
};

export default ChipButton;
