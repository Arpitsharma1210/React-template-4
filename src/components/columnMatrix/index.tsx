import React from "react";
import {
  ColumnsContainer,
  Column,
  IndividualField,
  FieldName,
  FieldValue,
} from "./styles";

interface Props {
  data: Record<string, any>;
  rows?: number;
}

const ColumnMatrix: React.FC<Props> = ({ data, rows = 3 }) => {
  let cols = Math.ceil(Object.keys(data).length / rows);
  let indexFrom = 0;
  let objectKeys = Object.keys(data);
  const updateIndex: any = () => {
    indexFrom += rows;
  };

  return (
    <ColumnsContainer>
      {[...Array(cols)].map((_, i) => {
        return (
          <Column key={i}>
            {objectKeys.slice(indexFrom, indexFrom + rows).map((item, j) => (
              <IndividualField key={j}>
                <FieldName>{item}</FieldName>
                <FieldValue>{data[item]}</FieldValue>
              </IndividualField>
            ))}
            {updateIndex()}
          </Column>
        );
      })}
    </ColumnsContainer>
  );
};

export default ColumnMatrix;
