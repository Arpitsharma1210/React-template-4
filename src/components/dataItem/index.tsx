import React from "react";
import { StyledDataItem } from "./styles";
import ActiveJobsIcon from "../../assets/images/activejobs.svg";
import ApplicantsIcon from "../../assets/images/applicants.svg";
import InterviewsIcon from "../../assets/images/interviews.svg";
import OfferedIcon from "../../assets/images/offered.svg";
import HiresIcon from "../../assets/images/hires.svg";
import StatIncrease from "../../assets/images/stat-increase.svg";
import StatDecrease from "../../assets/images/stat-decrease.svg";
import messages from "../../messages";

interface Props {
  type: "activejobs" | "applicants" | "interviews" | "offered" | "hires";
  data: string;
  stattype?: "increase" | "decrease" | "none";
  statdata?: string;
}

const ItemData = {
  activejobs: { title: "Active Jobs", icon: ActiveJobsIcon },
  applicants: { title: "Applicants", icon: ApplicantsIcon },
  interviews: { title: "Interviews", icon: InterviewsIcon },
  offered: { title: "Offered", icon: OfferedIcon },
  hires: { title: "Hires", icon: HiresIcon },
};

const DataItem: React.FC<Props> = ({
  type,
  data,
  stattype = "none",
  statdata,
}) => {
  return (
    <StyledDataItem>
      <div className="imgContainer">
        <img src={ItemData[type].icon} alt="dataitem icon" />
      </div>
      <div className="databox">
        <div className="title">{ItemData[type].title}</div>
        <div className="data">{data}</div>
        {stattype !== "none" && (
          <div className={`stat ${stattype === "increase" ? "inc" : "dec"}`}>
            <span>
              {stattype === "increase" ? (
                <img src={StatIncrease} alt="" />
              ) : (
                <img src={StatDecrease} alt="" />
              )}{" "}
              {(statdata ? statdata : "") + "%"}{" "}
            </span>
            <span>{messages?.dataitem?.this}</span>
            <span>{messages?.dataitem?.month}</span>
          </div>
        )}
      </div>
    </StyledDataItem>
  );
};

export default DataItem;
