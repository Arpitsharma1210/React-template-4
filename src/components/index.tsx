export { default as OnlyWith } from "./onlyWith";

/* FORMS */
export * from './form';

/* INPUTFIELDS */
export { default as Textfield } from "./inputFields/materialTextInput";
export { default as Autocompletefield } from "./inputFields/materialAutocompleteInput";
export { default as Selectfield } from "./inputFields/select";
export { default as Timefield } from "./inputFields/time";

/* HEADER */
export { default as Header } from "./layout/header";

/* Card */
export { default as Card } from "./card";
export { default as CardHeader } from "./card/CardHeader";
export { default as CardTitle } from "./card/CardTitle";
export { default as CardActions } from "./card/CardActions";
export { default as CardContent } from "./card/CardContent";

export { default as Button } from "./button";
export { default as ChipButton } from "./chipButton";
export { default as Dropdown } from "./dropdown";
export { default as InputLabel } from "./label";
export { default as Switch } from "./switchInput";
export { default as DataItem } from "./dataItem";
export { default as Table } from "./table";
export { default as Tooltip } from "./tooltip";
export { default as ModalWrapper } from "./modal";
export { default as Notifications } from "./notifications";
export { default as Sidebar } from "./layout/sidebar";
export { default as ColumnMatrix } from "./columnMatrix";
export { default as CustomAccordion } from "./accordion";
export { default as Toast } from "./toast";
export { default as NameAvatar } from "./nameAvatar";
export { default as Loader } from "./loader";


export * from './layout';
export {default as ListHeader} from './listHeader';
export {default as MaterialTextInput} from './inputFields/materialTextInput';
export {default as TextInput} from './textInput';
export {default as PasswordTextInput} from './passwordTextInput';
export {default as MaterialAutocompleteInput} from './inputFields/materialAutocompleteInput';
export {default as PopupError} from './popupError';
export {default as MuiCheckBox} from './checkbox';
export {default as Modal} from './modal';
export {default as ModalAction} from './modalAction';
export {default as Chip} from './chip/chip';
export {default as ChipGroup} from './chip/chipGroup';
export {default as Info} from './info';
export {default as TruncatedText} from './truncatedText';
export {default as Accordion} from './accordion';
export {default as SwitchInput} from './switchInput';