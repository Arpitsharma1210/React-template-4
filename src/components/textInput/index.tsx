
import React from 'react';
import { StyledContainer, StyledError, StyledInput, StyledInputContainer, StyledInputText, StyledLabel } from './styles';

interface Props {
    label?: string;
    value?: string;
    onChange?: any;
    error?: string;
    disableErrorMode?: boolean;
    required?: boolean;
    readOnly?: boolean;
    disabled?:boolean;
    onReadOnlyCtaClick?:()=>void;
}


const TextInput: React.FC<Props> = ({
    label,
    error,
    value,
    onChange,
    disableErrorMode,
    readOnly,
    required,
    disabled,
    onReadOnlyCtaClick,
    ...props
}) => {
    return (
        <StyledContainer>
            {label && (
                <StyledLabel 
                    readOnly={readOnly}
                    required={required}
                >
                    {label}
                </StyledLabel>
            )}
            <StyledInputContainer data-testid="inputContainer">
                {readOnly ? (
                    <StyledInputText 
                        onClick={onReadOnlyCtaClick}
                        disabled={disabled} 
                        variant='body2'
                        data-testid="readOnlyInputField"
                    >
                        {value || ''}
                    </StyledInputText>
                ) : (
                    <StyledInput
                    data-testid="inputField"
                        {...props}
                        disabled={disabled}
                        error={disableErrorMode ? false : !!error}
                        value={value || ''}
                        onChange={(event) => {
                            if (onChange) {
                                onChange(event?.currentTarget?.value);
                            }
                        }}
                    />)}
            </StyledInputContainer>
            {!disableErrorMode && <StyledError variant='body2'>{error || ''}</StyledError>}
        </StyledContainer>
    )

}

export default TextInput;