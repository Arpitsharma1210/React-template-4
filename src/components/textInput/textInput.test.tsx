import { fireEvent, render, screen } from "@testing-library/react";
import { TextInput } from "../../components";
import React from "react";

// test ids
const inputContainer = "inputContainer";
const readOnlyInputField = "readOnlyInputField";
const inputField = "inputField";

describe("Text Input Component", () => {
  it("renders without crashing", () => {
    render(<TextInput />);
    expect(screen.getByTestId(inputContainer)).toBeInTheDocument();
  });

  it("renders label through props", () => {
    render(<TextInput label="Email Address" />);
    expect(screen.getByText("Email Address")).toBeInTheDocument();
  });

  it("renders error message through props", () => {
    render(<TextInput error="An error occurred" />);
    expect(screen.getByText("An error occurred")).toBeInTheDocument();
  });

  it("runs onChange function passed through props", () => {
    const mockFunction = jest.fn();
    render(<TextInput onChange={mockFunction} />);
    fireEvent.input(screen.getByTestId(inputField), {
      target: { value: "sometext" },
    });
    expect(mockFunction).toHaveBeenCalled();
  });

  it("disables showing error through disableErrorMode prop", () => {
    render(<TextInput error="some error" disableErrorMode />);
    expect(screen.queryByText("some error")).toBeNull();
  });

  it("calls readOnlyCtaClick when it is readonly", () => {
    const mockFunction = jest.fn();
    render(<TextInput readOnly onReadOnlyCtaClick={mockFunction} />);
    fireEvent.click(screen.getByTestId(readOnlyInputField));
    expect(mockFunction).toHaveBeenCalled();
  });
});