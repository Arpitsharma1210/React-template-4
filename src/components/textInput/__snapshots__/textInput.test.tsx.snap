// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input Component matches snapshot when given props 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh hxIJRe"
  >
    <label
      class="sc-gtsqUy iUCbTq"
      readonly=""
    >
      Email Address
    </label>
    <div
      class="sc-dlniIP jcmohT"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 sc-eCAqax fJJTNW css-e784if-MuiTypography-root"
      >
        example@email.com
      </p>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 sc-jSFipO lhIbVb css-e784if-MuiTypography-root"
    >
      An error occurred
    </p>
  </div>
</DocumentFragment>
`;
